src/main/java/com/maarten551/tictactoe_backend/controller/LobbyController.java

```diff
@@ -0,0 +1,147 @@
+    }
+
+    @MessageMapping({"/send/lobbies/join"})
+    @SendTo(LOBBY_OVERVIEW_ENDPOINT)
+    public List<Lobby> onJoinLobby(@Payload String lobbyId, SimpMessageHeaderAccessor headerAccessor) throws LobbyDoesNotExistException, PlayerAlreadyInLobbyException {
+        Player player = this.playerContainer.getPlayerBySessionId(headerAccessor.getSessionId());
+        Lobby lobby = this.lobbyContainer.getLobbyById(UUID.fromString(lobbyId));
+
+        if (this.lobbyContainer.getLobbyByPlayer(player).isPresent()) {
+            throw new PlayerAlreadyInLobbyException(player, lobby);
+        }
+
+        lobby.players.add(player);
+
+        return this.lobbyContainer.getAllWaitingLobbies();
+    }
+
+    @MessageMapping({"/send/lobby/{lobbyId}/start-game"})
+    @SendTo(LOBBY_OVERVIEW_ENDPOINT)
+    public List<Lobby> onLobbyStartGame(@DestinationVariable String lobbyId, SimpMessageHeaderAccessor headerAccessor) throws LobbyDoesNotExistException, GameSessionHasAlreadyStartedException, PlayerIsNotLeaderOfGameException, NotEnoughPlayersInLobbyToStartException, GameSessionNotFoundException {
+        UUID lobbyUUID = UUID.fromString(lobbyId);
+        Player player = this.playerContainer.getPlayerBySessionId(headerAccessor.getSessionId());
+        Lobby lobby = this.lobbyContainer.getLobbyById(lobbyUUID);
+
+        this.lobbyContainer.startGameInLobby(lobby, player);
+
+        SelectedLobbyOverview selectedLobbyOverview = new SelectedLobbyOverview();
+        selectedLobbyOverview.lobby = lobby;
+        selectedLobbyOverview.lobby.gameSession.currentPlayingPlayerBySessionId = this.gameSessionContainer.determinePlayingPlayerInSession(lobbyUUID).getSessionId();
+        template.convertAndSend(LOBBY_JOINED_ENDPOINT.replace("{lobbyId}", lobbyId), selectedLobbyOverview);
+
+        return this.lobbyContainer.getAllWaitingLobbies();
+    }
+
+    @MessageMapping({"/send/lobby/{lobbyId}/execute-move"})
+    public void onPlayerActionExecution(@DestinationVariable String lobbyId, @Payload PlayerExecutionMove move, SimpMessageHeaderAccessor headerAccessor) throws GameSessionNotFoundException, WrongPlayerTryingToExecuteAnActionException, LobbyDoesNotExistException, FieldIsAlreadySetException {
+        UUID lobbyUUID = UUID.fromString(lobbyId);
+        Player playerAllowedToMakeAMove = this.gameSessionContainer.determinePlayingPlayerInSession(lobbyUUID);
+        if (!playerAllowedToMakeAMove.getSessionId().equals(headerAccessor.getSessionId())) {
+            throw new WrongPlayerTryingToExecuteAnActionException("It's not your turn!");
+        }
+
+        Lobby lobby = this.lobbyContainer.getLobbyById(lobbyUUID);
+        Player player = playerContainer.getPlayerBySessionId(headerAccessor.getSessionId());
+        SelectedLobbyOverview selectedLobbyOverview = new SelectedLobbyOverview();
+
+        try {
+            this.gameSessionContainer.executeMove(lobby, player, move.x, move.y);
+        } catch (GameSessionEndsInATieException e) {
```

Comment:
I think the concept of Exceptions is misused for these game states.

Reply:
This could indeed be solved differently, but what's wrong with using exceptions for this behavior?

You can use exceptions for more than errors 😉.